alcs-frontend/src/app/shared/utils/api-date-formatter.ts

```diff
@@ -0,0 +1,4 @@
+import * as dayjs from 'dayjs';
+
+export const formatDateForApi = (date: Date | number) =>
+  date ? dayjs(date).startOf('day').add(12, 'hours').valueOf() : date;
```

Comment:
Black magic?

Reply:
Yes, since we have to give everything a time (javascript has no concept of date without time) we move it to the middle of the day to keep things a little more standard than using the exact time from the user.

Reply:
Lets through that in as a comment for posterity

Reply:
Good idea, this would be very confusing for anyone to randomly see. Added a comment explaining this is done to "standardize" the dates.